### ISSUE TITLE
allow switching to custom software environment in Project Settings

### ISSUE THREAD

**Issue by DrXyzzy** [sagemathinc/cocalc#4053]
At present the "Selected image" menu in "Project Settings" > "Project control" does not include custom software environments. This means that it's not possible to switch a project to a CSE after it is created.

See also: #4052.

**Comment by haraldschilly** [comment]
IMHO that makes no sense to do. As a stretch, I can only imagine doing such a change once from one of cocalc's images to a custom one, but then it's locked in.

**Comment by DrXyzzy** [comment]
@haraldschilly Yes, I understand, after reviewing the interface more and reading your comments. It only makes sense to set CSE at time of project creation. I'm changing this issue from "backend" to "documentation" and assigning to myself.

**Comment by haraldschilly** [comment]
In the context of a course, I fully see the case for switching it later on. As of now, the first time the project starts  the accompanying files for the custom image content are copied over.  Maybe this is actually a shortcoming and shouldn't happen. I.e. only the files the teachers are distributing should be in the student projects. So, that might be another ticket. Despite that, today I spent some time on #3133 … but only moderate progress.
